src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -927,6 +929,14 @@ private Conversion ClassifyImplicitBuiltInConversionFromExpression(BoundExpressi
                     }
                     break;
 
+                case BoundKind.UnconvertedAddressOfOperator when destination is FunctionPointerTypeSymbol funcPtrType:
```

Comment:
What happens when the destination is `void*`?

Reply:
Will add a test, but this won't work. Like delegates, you cannot use `var` (or the equivalent).

Reply:
The type `void*` though is not the equivalent of `var`. This is an explicit type vs. asking the compiler to infer a type.

I think we're making a mistake in how closely we're taking this to method group conversions. The problem with method group conversions is that they must convert to `delegate` instances which are nominal types. Hence from the language perspective there is no right answer here for the delegate type of a method group conversion. Or rather there are 0 to N correct answers and the language decided it was simpler to say there was no type. 

This is not the same for function pointers. The typing is not nominal here and for a given method there is an un-ambiguous type for the `&` operation. The type is entirely structural and hence doesn't have the nominal problems of traditional delegates.

Given you're already proceeding on this route though and it's a subset of the feature I think we should table the discussion, add a prototype comment / note to chat with LDM to follow up.

Reply:
@jaredpar I think you make a good point. This is implementing the draft specification, but I agree the draft specification should be reconsidered.